src/main/java/com/powsybl/sld/server/PositionDiagramLabelProvider.java

```diff
@@ -92,18 +92,17 @@ static Integer getInjectionOrder(ConnectablePosition<?> position) {
     }
 
     static Integer getBranchOrder(ConnectablePosition<?> position, VoltageLevel voltageLevel, Branch<?> branch, boolean throwException) {
-        Integer order = null;
         if (branch.getTerminal1().getVoltageLevel() == voltageLevel) {
-            order = position.getFeeder1().getOrder().orElse(null);
-        } else if (branch.getTerminal2().getVoltageLevel() == voltageLevel) {
-            order = position.getFeeder2().getOrder().orElse(null);
-        } else {
-            LOGGER.error("Given voltageLevel {} not found in terminal 1 and terminal 2 of branch", voltageLevel.getId());
-            if (throwException) {
-                throw new PowsyblException(String.format("Given voltageLevel %s not found in terminal 1 and terminal 2 of branch", voltageLevel.getId()));
-            }
+            return position.getFeeder1() != null ? position.getFeeder1().getOrder().orElse(null) : null;
         }
-        return order;
+        if (branch.getTerminal2().getVoltageLevel() == voltageLevel) {
+            return position.getFeeder2() != null ? position.getFeeder2().getOrder().orElse(null) : null;
+        }
+        LOGGER.error("Given voltageLevel {} not found in terminal 1 and terminal 2 of branch", voltageLevel.getId());
+        if (throwException) {
+            throw new PowsyblException(String.format("Given voltageLevel %s not found in terminal 1 and terminal 2 of branch", voltageLevel.getId()));
+        }
+        return null;
```

Comment:
```suggestion
    static Integer getBranchOrder(ConnectablePosition<?> position, VoltageLevel voltageLevel, Branch<?> branch, boolean throwException) {
        Optional<ConnectablePosition.Feeder> feeder = Optional.empty();
        if (branch.getTerminal1().getVoltageLevel().getId().equals(voltageLevel.getId())) {
            feeder = Optional.ofNullable(position.getFeeder1());
        } else if (branch.getTerminal2().getVoltageLevel().getId().equals(voltageLevel.getId())) {
            feeder = Optional.ofNullable(position.getFeeder2());
        } else {
            LOGGER.error("Given voltageLevel {} not found in terminal 1 and terminal 2 of branch", voltageLevel.getId());
            if (throwException) {
                throw new PowsyblException(String.format("Given voltageLevel %s not found in terminal 1 and terminal 2 of branch", voltageLevel.getId()));
            }
        }
        return feeder.flatMap(ConnectablePosition.Feeder::getOrder).orElse(null);
    }

    static Integer get3wtOrder(ConnectablePosition<?> position, VoltageLevel voltageLevel, ThreeWindingsTransformer twt, boolean throwException) {
        Optional<ConnectablePosition.Feeder> feeder = Optional.empty();
        if (twt.getLeg1().getTerminal().getVoltageLevel().getId().equals(voltageLevel.getId())) {
            feeder = Optional.ofNullable(position.getFeeder1());
        } else if (twt.getLeg2().getTerminal().getVoltageLevel().getId().equals(voltageLevel.getId())) {
            feeder = Optional.ofNullable(position.getFeeder2());
        } else if (twt.getLeg3().getTerminal().getVoltageLevel().getId().equals(voltageLevel.getId())) {
            feeder = Optional.ofNullable(position.getFeeder3());
        } else {
            LOGGER.error("Given voltageLevel {} not found in leg 1, leg 2 and leg 3 of ThreeWindingsTransformer", voltageLevel.getId());
            if (throwException) {
                throw new PowsyblException(String.format("Given voltageLevel %s not found in leg 1, leg 2 and leg 3 of ThreeWindingsTransformer", voltageLevel.getId()));
            }
        }
        return feeder.flatMap(ConnectablePosition.Feeder::getOrder).orElse(null);
    }

```

Reply:
We could fix the get3wtOrder also in this PR, but we haven't developed the modification for 3WT yet, nor its connectivity
you suggested using Optional to handle the feeder nullability, yes, why not ? I'll go with that.